core/types/block.go

```diff
@@ -384,6 +384,9 @@ func NewBlockWithHeader(header *block.Header) *Block {
 // CopyHeader creates a deep copy of a block header to prevent side effects from
 // modifying a header variable.
 func CopyHeader(h *block.Header) *block.Header {
+	if h == nil {
```

Comment:
I thought about doing this as well but it’s ick to give back nil with no reason but actually I think some more and in this case it’s fine since caller gave nil